### ISSUE TITLE
`$props.id()` false positive "possibly undefined" when props type has an optional `id`

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#171]
`svelte-check-rs` 0.11.0 reports `$props.id` as possibly undefined when the component's props type declares an optional `id?`. It appears to resolve `$props.id` to that prop rather than to the [`$props.id()` rune](https://svelte.dev/docs/svelte/$props#$props.id()).

This hits any component spreading `HTMLAttributes`, since those carry `id?: string`.

**Repro** (svelte 5.56.8, svelte-check-rs 0.11.0):

```svelte
<script lang="ts">
  let { name, id }: { name: string; id?: string } = $props()
  const uid = $props.id()
</script>

<div id={id ?? uid}>{name}</div>
```

**Actual:**

```
Error: Cannot invoke an object which is possibly 'undefined'. (ts(TS2722))
Error: '$props.id' is possibly 'undefined'. (ts(TS18048))
```

**Expected:** no errors. Official `svelte-check` (TS 5.9.3) reports 0 errors on this same file, and Svelte declares the rune as non-optional: `declare namespace $props { export function id(): string }`.

**Narrowing** — only the optional `id?` triggers it:

| Props type | Result |
|---|---|
| `{ name: string }` | pass |
| `{ name: string; id: string }` | pass |
| `{ name: string; id?: string }` | **2 errors** |
| `HTMLAttributes<HTMLDivElement>` | **2 errors** |

Workaround: `const uid = ($props.id as () => string)()`.
